### PR TITLE
Cr 1069 modernise dependencies - and update structure a little

### PR DESCRIPTION
**Pull request by andrewjohnysons**
Upgraded Cucumber dependencies
Upgraded Spring / cucumber dependencies
Removed inheritance from Spring runner
Glued Spring runner
Moved glue classes to their own package
Generally brought everything into line with RH-CUC, so they should broadly match in structure

### CODE REVIEW

**Comment by andrewjohnysons** [comment]
Passed in dev - but coughed a bit when it tried to connect to EQ the first time round

File: pom.xml [hunk 1]
```diff
@@ -50,17 +50,14 @@
...
-
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-web</artifactId>
     </dependency>
 
     <dependency>
-      <groupId>com.jayway.jsonpath</groupId>
-      <artifactId>json-path</artifactId>
+      <groupId>org.springframework.boot</groupId>
```
Comment (rpcatling) [inline pom.xml:57]: We seem to have a duplicate of this dependency now - please remove one of them.

File: pom.xml [hunk 2]
```diff
@@ -138,10 +135,8 @@
           <!-- <testClassesDirectory>census-rh-cucumber/src/test/java</testClassesDirectory> -->
```
Comment (rpcatling) [inline pom.xml]: why not remove this commented out line while you are here ?

File: src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/glue/RunCucumberTestSSO.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.ons.ctp.integration.fieldsvccucumber.glue;
+
+import io.cucumber.junit.Cucumber;
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+    plugin = {"pretty", "html:target/cucumber"},
+    features = {"src/test/resources/features/sso"},
```
Comment (rpcatling) [inline src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/glue/RunCucumberTestSSO.java]: while you are here , and renaming directories, would it be possible to remove the annoying "sso" folder under features ?  Would also need to change the directory for the actual feature file!

File: src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java [hunk 4]
```diff
@@ -2,9 +2,11 @@
 
 import lombok.Data;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
 
 @Data
 @ConfigurationProperties("config")
+@Component
```
Comment (rpcatling) [inline src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java]: unnecessary change. The ConfigurationProperties annotation is enough for Spring.

**Review by rpcatling — CHANGES_REQUESTED** [review]
Just a few comments

File: pom.xml [hunk 5]
```diff
@@ -26,21 +26,21 @@
...
-      <version>4.4.0</version>
+      <version>5.4.0</version>
       <scope>test</scope>
     </dependency>
 
     <dependency>
       <groupId>io.cucumber</groupId>
       <artifactId>cucumber-spring</artifactId>
-      <version>4.4.0</version>
+      <version>5.4.0</version>
```
Comment (MartinHumphrey) [inline pom.xml:43]: There are later versions of cucumber-java, cucumber-junit and cucumber-spring if updating to latest 6.8.1 Oct 2020?

File: pom.xml [hunk 6]
```diff
@@ -50,17 +50,14 @@
...
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-web</artifactId>
     </dependency>
 
     <dependency>
-      <groupId>com.jayway.jsonpath</groupId>
-      <artifactId>json-path</artifactId>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
```
Comment (MartinHumphrey) [inline pom.xml:58]: It appears the later dependency <artifactId>spring-boot-test</artifactId> should actually be included as part of <artifactId>spring-boot-starter-test</artifactId> and so could also be removed?
https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-test-scope-dependencies

**Review by MartinHumphrey — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (andrewjohnysons) [reply pom.xml:57]: done

File: src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/glue/RunCucumberTestSSO.java (hunk 3 above)
Comment (andrewjohnysons) [reply src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/glue/RunCucumberTestSSO.java]: Don't know why we have this - to be honest

File: src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java (hunk 4 above)
Comment (andrewjohnysons) [reply src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java]: Meh

File: pom.xml (hunk 5 above)
Comment (andrewjohnysons) [reply pom.xml:43]: I know, but we are currently using 5.4 across the board so I want to bring them all into line so all cuc projects are using the same dependency version then upgrade them all together, if that makes sense

File: pom.xml (hunk 6 above)
Comment (andrewjohnysons) [reply pom.xml:58]: done

File: pom.xml (hunk 2 above)
Comment (andrewjohnysons) [reply pom.xml]: I think it's wrong anyway - shouldn't be RH - should be 
<testSourceDirectory>census-field-service-cucumber/src/test/java</testSourceDirectory>

File: pom.xml (hunk 1 above)
Comment (andrewjohnysons) [reply pom.xml:57]: I removed it completely actually

File: src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java (hunk 4 above)
Comment (andrewjohnysons) [reply src/test/java/uk/gov/ons/ctp/integration/fieldsvccucumber/cucSteps/sso/TestConfig.java]: It does work without it but my jury is still out :-)

**Review by rpcatling — APPROVED** [review]
